release/benchmark-worker-startup/benchmark_worker_startup.py

```diff
@@ -0,0 +1,363 @@
+                metrics_actor_name,
+                metrics_actor_namespace,
+                test,
+            )
+        )
+
+    measure_disk_speed()
+
+@ray.remote(num_cpus=0)
+class MetricsActor:
+    """
+    Actor which tests will report metrics to.
+    """
+
+    def __init__(self, expected_measurements_per_test: int):
+        self.measurements = defaultdict(list)
+        self.expected_measurements_per_test = expected_measurements_per_test
+
+    def submit(self, test_name: str, latency: float):
+        print(f"got latency {latency} s for test {test_name}")
+        self.measurements[test_name].append(latency)
+        results = self.create_results_dict_from_measurements(
+            self.measurements, self.expected_measurements_per_test
+        )
+        safe_write_to_results_json(results)
+
+        assert (
+            len(self.measurements[test_name]) <= self.expected_measurements_per_test
+        ), (
+            f"Expected {self.measurements[test_name]} to not have more elements than "
+            f"{self.expected_measurements_per_test}"
+        )
+
+    @staticmethod
+    def create_results_dict_from_measurements(
+        measurements, expected_measurements_per_test
+    ):
+        results = {}
+        perf_metrics = []
+
+        for test_name, measurements in measurements.items():
+            test_summary = {
+                "measurements": measurements,
+            }
+
+            if len(measurements) == expected_measurements_per_test:
+                median = statistics.median(measurements)
+                test_summary["p50"] = median
+                perf_metrics.append(
+                    {
+                        "perf_metric_name": f"p50.{test_name}",
+                        "perf_metric_value": median,
+                        "perf_metric_type": "LATENCY",
+                    }
+                )
+
+            results[test_name] = test_summary
+
+        results["perf_metrics"] = perf_metrics
+        return results
+
+
+def print_disk_config():
+    print("Getting disk sizes via df -h")
+    subprocess.check_call("df -h", shell=True)
+
+def measure_disk_speed():
+    subprocess.check_call("sudo hdparm -t /dev/nvme0n1", shell=True)
+
+def generate_test_matrix(
+    num_cpus_in_cluster: int,
+    num_gpus_in_cluster: int,
+    num_tasks_or_actors_per_run: int,
+    num_measurements_per_test: int,
+):
+
+    num_repeated_jobs_or_runs = num_measurements_per_test
+    total_num_tasks_or_actors = num_tasks_or_actors_per_run * num_repeated_jobs_or_runs
+
+    num_jobs_per_type = {
+        "cold_start": num_repeated_jobs_or_runs,
+        "warm_start": 1,
+    }
+
+    imports_to_try = ["torch", "none"]
+
+    tests = set()
+
+    for with_tasks in [True, False]:
+        for with_gpu in [True, False]:
+            for with_runtime_env in [True, False]:
+                for import_to_try in imports_to_try:
+                    for num_jobs in num_jobs_per_type.values():
```

Comment:
I think we can simplify this a little bit (and match what is specified in doc)

-the first run is cold start, where the cluster has not accepted any jobs 
-the subsequent runs are warm start

the workload should be identical in both runs as far as the number of tasks goes 

We should see bigger difference between the two if we track it this way